code/modules/projectiles/boxes_magazines/external_mag.dm

```diff
@@ -346,13 +346,20 @@
 
 /obj/item/ammo_box/magazine/toy/pistol
 	name = "foam force pistol magazine"
-	icon_state = "9x19p"
+	icon_state = "9x19p-8-toy"
 	max_ammo = 8
 	multiple_sprites = 2
 
 /obj/item/ammo_box/magazine/toy/pistol/riot
 	ammo_type = /obj/item/ammo_casing/caseless/foam_dart/riot
 
+/obj/item/ammo_box/magazine/toy/pistol/update_icon()
+	..()
+	if(ammo_count())
```

Comment:
You can one line this with
`icon_state = "9x19-[ammo_count() ? "8" : "0"]-toy"`